validator/coordinator/src/main/java/tech/pegasys/teku/validator/coordinator/BroadcastAttestationsHandler.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.validator.coordinator;
+
+import com.google.common.primitives.UnsignedLong;
+import tech.pegasys.teku.sync.SyncService;
+import tech.pegasys.teku.validator.api.BroadcastAttestationsChannel;
+
+public class BroadcastAttestationsHandler {
+
+  private BroadcastAttestationsChannel attestationChannel;
+  private SyncService syncService;
+  private volatile UnsignedLong lastBroadcastSingleAttestationSlot = UnsignedLong.ZERO;
+
+  public BroadcastAttestationsHandler(
+      BroadcastAttestationsChannel channel, SyncService syncService) {
+    this.attestationChannel = channel;
+    this.syncService = syncService;
+  }
+
+  public synchronized void broadcastSingleAttestations(UnsignedLong slot) {
+    if (slot.compareTo(lastBroadcastSingleAttestationSlot) <= 0 || syncService.isSyncActive()) {
+      return;
+    }
```

Comment:
I feel like the protection to avoid broadcasting attestations for the same slot twice should wind up in the validator client rather than the beacon chain.  That may be particularly important once its split out to a separate process if the connection winds up failing over to a different beacon chain node for some reason.

So I'd have probably had the same events coming out of the BeaconChainService but publish attestations early in `DutyScheduler` and have this check to ensure we don't perform the duty twice there. This same check would also probably be good to add for block production to make it even less likely we wind up doing that twice.

Reply:
This actually makes a lot of sense to me. I'm working on this right now and it turns out to be way more simpler too.

Reply:
Since this new method of doing this was way simpler, I started with a clean slate and force pushed. FYI.